File: 8/Java/superclass_pointer.java
import java.io.*;
import java.lang.*;

class a
{
	void g()
	{
		System.out.println("anil");
	}
}
class b extends a
{
	void g()
	{
		System.out.println("hari");
	}
}

class kapil
{
	public static void main(String[] args) {
		a x;
		b y;
		y = new b();
		x = y;
		x.g();
	}
}

File: 8/C++/superclass_pointer.cpp
#include <iostream>
using namespace std;

class a
{
	public: void g()
	{
		cout<<"anil"<<endl;
	}
};

class b: public a
{
	public: void g(int w){cout<<"hari"<<endl;}
};

int main()
{
	a *x;
	b y;
	x = &y;
	(*x).g();
}